pygmt/gridops.py

```diff
@@ -113,3 +113,115 @@ def grdcut(grid, **kwargs):
+@use_alias(
+    G="outgrid",
+    F="filter",
+    D="distance",
+    I="increment",
+    N="nans",
+    R="region",
+    T="toggle",
+    V="verbose",
+    f="colinfo",
+)
+@kwargs_to_strings(R="sequence")
+def grdfilter(grid, **kwargs):
+    """
+    filter a grid file in the time domain using one of the selected convolution
+    or non-convolution isotropic or rectangular filters and compute distances
+    using Cartesian or Spherical geometries. The output grid file can optionally
+    be generated as a sub-region of the input (via *region*) and/or with new increment
+    (via *spacing*) or registration (via *toggle*).
+    In this way, one may have “extra space” in the input data so that the edges
+    will not be used and the output can be within one half-width of the input edges.
+    If the filter is low-pass, then the output may be less frequently sampled than the input.
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+        outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    filter : str
+        Name of filter type you which to apply, followed by the width
+        b: Box Car; c: Cosine Arch; g: Gaussian; o: Operator; m: Median;
+        p: Maximum Likelihood probability; h: histogram
+        Example: F='m600' for a median filter with width of 600
+    distance : str
+        Distance flag, that tells how grid (x,y) rrlated to the filter width as follows:
+        flag = p: grid (px,py) with width an odd number of pixels; Cartesian distances.
+
+        flag = 0: grid (x,y) same units as width, Cartesian distances.
+
+        flag = 1: grid (x,y) in degrees, width in kilometers, Cartesian distances.
+
+        flag = 2: grid (x,y) in degrees, width in km, dx scaled by cos(middle y), Cartesian distances.
+
+        The above options are fastest because they allow weight matrix to be computed only once.
+        The next three options are slower because they recompute weights for each latitude.
+
+        flag = 3: grid (x,y) in degrees, width in km, dx scaled by cosine(y), Cartesian distance calculation.
+
+        flag = 4: grid (x,y) in degrees, width in km, Spherical distance calculation.
+
+        flag = 5: grid (x,y) in Mercator -Jm1 img units, width in km, Spherical distance calculation.
+
+    increment : str
+        x_inc [and optionally y_inc] is the grid spacing.
+        (http://docs.generic-mapping-tools.org/latest/grdfilter.html#i)
+    nans : Str or Number
+        Determine how NaN-values in the input grid affects the filtered output.
+        Values are i|p|r (http://docs.generic-mapping-tools.org/latest/grdfilter.html#n)
+    {R}
+    toggle : Bool
+        Toggle the node registration for the output grid so as to become the opposite of the input grid
+        (http://docs.generic-mapping-tools.org/latest/grdfilter.html#t)
+    {V}
+    {f}
```

Comment:
**-f** is a GMT common option. We haven't added it to PyGMT yet. Currently, we can remove `{f}` first, and maybe add the **-f** alias in another PR.
```suggestion
```

Reply:
Yes, agree to remove `-f` for now until we wrap an alias ('colinfo'?) for it (e.g. following `-x` at https://github.com/GenericMappingTools/pygmt/pull/625). I would have considered it low priority in the Python world, but the `-f` flag has been mentioned recently at https://github.com/GenericMappingTools/pygmt/issues/597#issuecomment-696996503.